File: src/main/go/leetcode/lists/hot100/LeetCode11.go
package hot100

func maxArea(height []int) int {

	if len(height) == 0 {
		return 0
	}
	res, l, r := 0, 0, len(height)-1
	for l < r {
		res = Max(res, Min(height[l], height[r])*(r-l))
		if height[l] < height[r] {
			l++
		} else {
			r--
		}
	}
	return res
}

File: src/main/java/leetcode/lists/hot100/LeetCode11.java
package leetcode.lists.hot100;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/15 23:03
 */
public class LeetCode11 {

    public int maxArea(int[] height) {
        if (height == null || height.length == 0) {
            return 0;
        }

        int left = 0;
        int right = height.length - 1;

        int max = Integer.MIN_VALUE;
        while (left < right) {

            int temp = Math.min(height[left], height[right]) * (right - left);
            max = Math.max(temp, max);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }

}
